src/document_add.c

```diff
@@ -168,6 +165,20 @@ int RS_AddDocument(RedisSearchCtx *sctx, RedisModuleString *name, const AddDocum
   }
   Document_LoadPairwiseArgs(&doc, opts->fieldsArray, opts->numFieldElems);
 
+  if (!exists) {
+    // If the document does not exist, remove replace/partial settings
+    addOptions &= ~(DOCUMENT_ADD_REPLACE | DOCUMENT_ADD_PARTIAL);
+  } else if (!(addOptions & DOCUMENT_ADD_PARTIAL)) {
+    // delete old decument if REPLACE without PARTIAL
+    RedisModuleKey *dk = RedisModule_OpenKey(sctx->redisCtx, name, REDISMODULE_WRITE);
+    if (dk) {
+      if (RedisModule_KeyType(dk) == REDISMODULE_KEYTYPE_HASH) {
+        RedisModule_DeleteKey(dk);
```

Comment:
Do we need to handle deletion like we do in `DeleteCommand()` i.e. with `RedisModule_Replicate()`?

Reply:
No, since the `FT.ADD` command calls `RedisModule_Replicate` at the end of execution.